### ISSUE TITLE
Incorrect Mjölner Trigger Rate when accuracy is not 100%

### ISSUE THREAD

**Issue by Pro7ech** [PathOfBuildingCommunity/PathOfBuilding#7562]
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate open **and closed** issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues?q=is%3Aissue)

### Check for support

- [X] I've checked that the behaviour is supposed to be supported. If it isn't please open a feature request instead (Red text is a feature request).

### What is the behaviour in-game?

Effective Source Trigger Rate is above the Trigger Rate Cap while it isn't in PoB.

### What is the behaviour in Path of Building?

PoB multiplies the Attack Rate by the Hit Chance before checking the Trigger Rate Cap.

This results in the following miscalculation: if the Hit Chance is not 100%, then a higher attack speed is suggested to reach the trigger cap. However, since the Hit Chance in game is not a a continuous value, but rather a probabilistic discrete value, this result in an effective in game Trigger Rate that gets higher than the Trigger Rate Cap.

### How to reproduce the issue

![image](https://github.com/PathOfBuildingCommunity/PathOfBuilding/assets/39432395/4ffef46e-e1e3-4170-a408-d70271abc1fe)

### Character build code

```shell
eNrtfWtz3Lix6OfMr2Cpam-dW5ZlvEn62jmlh_XYlWytZK938yUFkqCGaw4pkxzL2pT_-22AHL5EzoykkZNNklScEQg0gEa_0QBe_e_XWWx9UVkepcnrLbyDtiyV-GkQJVevtz68P3zubP3vXyevzmUxfRfuzaNYf_nr5C-vzG8rVl9U_HrLdbasQmZXqvhlAYr-HUBdy6SYqjQ5k7-n2VEavN46nScyi_Ity5NJEBWvt96midqy_Fjm-Vs5U6-33qvZdSyzLUvmvkqC_ebLcaSy9HoKMKHBVGbSL1R2qkewOy_SszSAOqGMc4A3k1FymfqfVHGUpfNrGM6W9SVSN2Wlk7PzdxfvW8OLkmp4RzKbf4IetmCKf3l1HstblV0WsrBy-Of11i5gSl6pAzmDfwGijOcAznYIc3ZsTKhgNsJbL5Y23ptnefEwCJfXSgV1I7bDHD5W9TxTb8JQ-UX0Re1nUbEPk_KbDgneGW87VB_tuLZY1uBsHhfRdQxLVDcancfxHfjuKOz3aSHjg_PLuirlyGV8h9nM5Vw4y9ulRd0OjdX8GBXTvRgw-4BedNuTqyQq1AMbn6dRniaPmF-76egU9-dxDHzb6cZxXbrjMiwEthkda3mhcpV9kUXUHeR4T-nMi5IeLnVX7s6qrnYzJd-FJdVeyCCa52eqyFTeEOHOaNszmcj9NG9Wmy6teq4ykC1FpwVa0eBS-SmIo3YTDEyLEF-jp-H2o12eRqFav-a9ZlM1uO9oHjaPN5fr1rs34IcN6AKk7no1L9N5vGbNImsJcmeckz63azJMxmoeqK8NiYllADs1-SghniQtmqWULYHYrkoRGh_il1TLhNXYMXLqzfF5MwCHMox3KLEJYsQeVWDT2zzyZXwmv0az-QxUxnv5STUdUpcSMU6xV9MiAYk31hrkEBqf3WGUqdGWjNrUHRd_cfCwPkEhpvlYU0EIx-NiJkqOwZra9f052ES3DYqw6yzj6Uaw2sRdYnP4L3Xtk8RviNddBvdDkhmt0bJVCF_W4AI4WBtHXqzWbNF0UcmBhhB3XCyQA3aUwwXlzvJ-r1RSdX67nhw8VcqfHgHGL2Sh1tMdLdZjgizFs67ewbNtL4M8gOilfXTb9DGH0dJpDCMLI0KXq-VhjBHhkp1RQfQmUdnV7eU0UnFrahxztE6LxSD35fV6jQ3u2xDaa-C66_U5hB3KhbveLIexhG2QlO4oJ3yUWbCezrsnpb_5IvO2dAc5TZZTbtWijbhxalJgbUODQPW8AIeOuzPp79qZie_ZbjebpfOsrSPtpdMo63dYEK3SUKUvd6GCud9RiXQUY3sxuKZ9D2sZfuN4qMnoyHaLQvqfDtLgSq3bxHRyrxa1e2maXs6vr0GwaBpYF4DWsuBeRC0by1mj8jsg4TZnC7pMGfc7YKsr3-nAXWlmrN1Lr0W_K8qXmge9bsZJplW738Xq9TwDITEDxWCiDWdpSwAvEUWH4GG2fT5MiIsd0A2MYzSukUyzYSe2gnj5KYpjK9f_nssMmmxZObCoDzXPfk_jRMcberi19sEfmEeFlYZWMVXWsZJfVJK3FR3GmPAdTEltha_ubqs1ZqvyqK2BOe_QzpzXB9xMAAR7s2I2uJg22mGuUzsX68Pcv_VjE2G7X3ThPL0BspnqkF6DN7ZObTBhVzc4zFTyx20fPlqrersDOq5aA7CFQQytPYd-i6F5vI9moLry_EAW0goqF-gXmUUyKYiOd1m5kpk_PQW-O5Rx7IEUNktRl5ZLpOOSh1FcqOwAynSnevJ9iHgRQXv1wgRb9a-T2XWaFVYs8wLM_XSeFMcyn77eCgPlIZcr32FuyKQijiBKuCEJbT90uO2y0As5trdM0wsl4xks2_6Wpb5qgJp6buvgqa6yv4ivlvBtbAubO0DflCqk3NBzWMD9gCjqASUhV_DAA9ovx2vAwXTyIkpM6AhUUhxvWZfT9GY3-KIn_D5N47zuUV5fqyRY_GlgvM-UsuRCwfgaF2YN9B_WDEYIhkwpsgDMPyiz8TZY-xh92_4Hp8zZxjBnAn8w5pJtDobYt05E-QR6I1tWksIoYXrUcbapYyO8LRh36DZwMtkmyHXsbSoEZdsEfCm2Lahri22OMXK2BdgSFPoENtqmlGK0TRxGoCYXHG-DxeTybYptBMBsrqEJ7IptZoMnty0ww-42w0RQAwdaEQbQsANohpquQNvM4Zhtl8PHLqHbnFMKYFwbE_jIoIQIYiMYN8fCgOHbGFHAgwsu0Da3McACWQRtGBVET8R2YAAU2gtGYMBCiwDdBUyHIRBa2wx8fvjX1VOgUA5wEXUBNzAumBSHCW2DmONo22aMQR8MRoEIcbY5FJJtlwEoDOKVm5HDvMDLhyowGMAehkpQIrieOyJCd4T4tq2ny8FnIxrNHMYiXI1OV1cBqHptYIFdGC-IQIMqwACsipkBzEmTJUwSMA8fEVSB_mECgG_KzTRsGJ6rP1IYGBGOgK4F0phl0AZTm0N7wB2spyEk7MDyYL16uk-3_Bfm6BCsxyKoxgUC9GIE0GE1DH5BssKoYZ2dbZcQGARi0A8MHOphRgAKQZqGmMGCS6ALYcOiAYrNFG2uRws2P6BYz4g6WFMbhbqangBJBNaUQtfCMR1RG0BS3T-B0THHtg0YPQrXdWF5uJ6p4wI7GJ4QCMFoMYe1ht8ai4AmzLSA0iE0md3udjkEhFIBfNja1CGs2qzRn7lhyb-8-nBxan78ZVoU1_nLFy9ubm52rmUxTUP1FWz2HT-dvbiGRsDMz426eq7BvtiF_-xdfdhXF3_48S11Lm5-Rx_C-flR_OvVc_X78fVJ9ulqyr_sH_38x_VtAYT_5eRv8SX79PzvWXrp_nr7AfTeL_vu1fTo7LY4_-3H459__nIZzJLrv6G9XeXZb93zg1-_hj-l7tH87YFAn908-V0kz8-O_4gOs8y-Pkvynz98IVP04XDmnu-doY8_nu0mV8J15Ke3n_zZzWn65nTv64Xz29fP_Jevmf_c_3j6xj85Pf7FifDebjH9Nd_7eLmPwven704-fr28-eW2SJ3s57PDX3bx_JeT64PfXOfyj7e3jHw--NXePf3p_LMj4qnc5Wn242fbPf3x5rfz83DPu4j3ps-_fLpy54fqjx-_xpcHPx8L_0RO57f-4bQ4OpTO50z8dOV8xT8VqTj58vMH_u6zON-n-bu3wU_7u7vx7v5vuz_e7F7tftzd_enqMNr9mB4cvru5-P3kSwTjODqcvv9kFuzFYsVelZtkebl81V9GKur1NbJwy4oKNdN_4kpB362o2b6pZ49VM6K0qUfG6hlh29RzRuFpTmyNbxSgljBNPTZWzXB8U4-Ojk-L8la_Y_WMmG_qibF6RhG04I0DBG5v6rmj49M6oqlXGe2w7O3VfqU9gywC_VcaNy-0ZjXWhla9-sfbtFCmri5c_PHKGJ05WDFZcaRm-d4tGKKHOoTR2_ypdLeufamK0jpqt3m9VWRzpa2eUM5jXf7zXMaRtkMIaheflnvMSZrN6nAwwAJbQjudJcj3t9daj--enpZfduOigqb7WxgWpQFRjciKgoVRURWaXePdZtj7MvZzM_Ao8eM54DapDO_F2GPp6aHpPXMdvAwW5Xmc6vCMktdpYuGtPmCYTBRvVWsHA6yaH8WpJ2NSfl1srWtcXJXreKYKGYDp-eIEFjZ_oWf2wsDTSFr4DpXvo21XY0Xqhv2PgJ1ft0q_Yfzz5_ZqVH_o2gflwmx1Bo0XM-8hwpio1YQSOatMuZWe2os1cYOX46Ymj10wvxcb7jVWFoVVtRZCqpKmVY0LvmlU1H2sOWW-5oxrFJ-rRBWZscLP43k-TBe9SqMYGYf6hNSye6P3QlRgNWTT6r6RbXqsq7nZAL_LzyWOl7LzsYpnqvgezHwZXUXxu9C41Z0F636oV2c_yvxY1eVPuBZmAJpby642xKfVtIEXZKL241R-6jNqU15PulP6lORn-rHKjl5sZp1rNj1J_EzJHFTmvCLn9sTvfB1lzAE4T4iSujer7m7TiCmxfjnP7grtRfkyud20rdHwVIRR9rMZKbSOVbGXBrfWYl_jjizCWxtlyVb4sF6CuqyRP4uSJ6S5RR-bNg3qLcrOHOvSUSJrtXs6EtOdWGUvG7YP9mVefJyCb653axJlgtl37IOxSqNIGYf6PewD3btlurda_W9cNFWh_8siiz71OKP9ZRxJ3fZPRz2LPYqqp01zjtnxHNFcvW-juLgD4ym9DpXnG1dYffdrsXNz18Yuv9x1usryet746Zwt09N3M5f30rTIH2Ut0_VUVAzILm67jLgoa1TUouQJ2a3qYtPyZi9O0-BMXkV-Z45N8Sh3tVs-qYWcpVdmz_9PRFxrMvZhluaFB2K065K0i2usn_jqQOZPahC0-v1uuF6EsazLG51H8Bicr-kHenkaz-_olHZx4wW2Cp-SxJtu1mXvdR2dIIg0XBn_et6dbuvDuKvTaf2ECHiTxFqPqOTfmO4uizQDw1jvVLdjLk1pg_2m7CmDLboXy3Tz3ZAO4_yivotkPcqk3zWfq5Iay9XfTydNyw42ra_fFCoDhtzTmSLa_2hPsfdt3LuE_8l4P02-RP6TC7dqUFY94s2G9fZkUcRKB7nz_axrp_U_1Wjof3jC2Tdd_R-QN_n_yy3d42bdA83D79PdbNb3r5vixlJtFT5lPAW6sYrUMh19b5lOvksU3aS0muTcXfD6ukK9_61hwP6XpxM_f1MSkLJxYvsIHm43mlWV1HOs_n66mZUd_FtS1UGU-_qcay8o2i6u8dwufDpkt3p5CowvMko1Ll4S9-XlVF6rbCEr36dzf7oNInOey0Bl1sJ-eCrjArfQ-yY2ecoyhvX_lIDu6mL5EQpxAdiqId_BbPXzUhUmZ6LMitA_DW5NjZPkel4YsK-3ZrBIf_fmYahPoOvk3MycsH9zePhm__3JL2-qDNN2EzPPvyfzmac39cr_bw5VXypjGFj53MvLn6-3fonUjRnIARBwFOd6fnEsr3NV51aarIdq5DG0WwLN1DqO6rPqw7CaCuOQ3nxVWVGGuvwsUqPjqr-vGFTZoT7LojNyx6Dpw9_jgMozEjpkXJ64GcGUOXE_DkWffB-djv64pO3J7FrGoz1XX1dgotB5L0CzURj5Opl5-ZLrLJmy1hK81KcIR9e7OuAxDsOcph8DUH4cb1yeiB9rXX1dglVzGn8Uq-XX8eYHypejcy8_jjeuD3GlybFOwRmGUtdaAultmhgiB6bZjWItjEZXFoRVXWUc4LtiqrIqX3oM0hnIqEWVpYyTRd68GGfjVo0luDIHUEcwpL-NNy0PVY7MQX9bIok6BwtHENquMw6qPHw3KsiWNS1P5ozirzrms2QJqt3fEfSXX5cgYXG6b2T-1eclTFKax1_SKCiPeo2wS6_aMoGR-p8eD8acX3s8mP6BtsdDPIxl_ml0vauv480_FJG2agaglFbMWkA0Uz0Oguatx0G46BsSTduL5SZEfY5nsPHi6zLGr473PBhCeQjpwc2r3KwHtjbiG3wFBTNYKr_rOkvIu5gnzYbro0CZYQ3LgWZ294JVarLBmd4bYsmf1U0Ky1i4rLICEKji4yXG3nqQ6kOWx-D769t_0vhxAO_cGPGoeaZFLpPgQB8Wf-RE9Vnz-TUAW4zs3ZC13ixpH-qrFws_ypwE055NdYzussi0W_pHms5-06nuuPx9Wu8Zm8xwsHIPIkB2Zohk0ZOu-OvrrefVQcpXxr-vMtX170Wi-jxX5a0yZWTBFLdSyHXVMn3cLt3WC7Pz-9L68Pbk5w9vJtXB1MmRhDFNPiTR57myTg5eWhIL6mPqYEKwbbvExgRTwjzpuUFgC8UJV4ogBv_zHNtRlBAig8B3AleGVPoT07OZ6kvLQZMqx_2lRdCkyu1_ae09h_9OTKUL9fmlJdAEnIk48iP9FU_sH7TPXyXwlW6QZfycyfssuroCJ15aJbBOzq3RihYsM6zotnUTFVOoh3YIr04PWftAzEF6k0zGGufWFBBiJamlL9axotCqOoSun1GErJOk0Dk7V3rNLBj7PMqMIz55RuArLLxKrqDb9heMaHs6izP5VklPE0e0vzbjqT5XZxlA_kaJ9QxbBRBoPtlPs2x-DRMwLj847BcyuVJWpv_VF4zwrersQBUUGK1BVtZgK2vwlTXEyhr2yhqLg8Am4tWlcUK6NH62e3SyPznXp0PN1mMTC7qMYmByy1gBi4z-s3med3kAc9_3qZS-i5BCIWLCl4yL0NNn-Xw_DCgnDg-FIoxgLjgD6rdt4BsR2FKGosMDnDc8gFo0T502zaMJQT9YWZlrZFXa1rqGsQKjTxj5wYrbuUHQdDYHP_FWx8dLj9EC60HTTSm_JxdqpiNZ1t48M-QEXKEhPZIY6ObIZWghKR5ayN3Yn6oZ-BvLVnF_qlQhp5P9TIbAFy8tLU6BBlQYfX1p_cMM4iX6ZtpUF7h1E7vo5HIetivjsvKeiYi17hsw6XtaOQJaaUe-tQRaZ3HJ5B9-OaxvNez_Eei5jf5vm_dVuXJ1Xe2A50YWVeTQXWFCG4Kp6YLwAYi4I08Xkyklag_o0Mrhf4Ko6FNHFcc22aOGUsgaaq06pf7SAjKwyA7aQf0idrdI9IvoDubtsv15lmm5XkYnAYP1Fz6s7S6ew39HtR0QRzW5l3ibbNNvDLVXS5vC9fJqLgbffxaBCqhb8W32DfOGFnSF27Ld-yn4h1PQeZMFA-zwb__joOeYdEmvr5SGtU53oNBtGypFz1kXaF-XlRp5oc5ND61ZfOsPES2HtkTjLtfUzSy-cUC1b0KFWo6a5NyVhkXTnnyjj2oPGNzm3zZk0dxZm3ETJwDZZQF6O2RmOqrWqWP2PLmBQf9sJsiSGu7KGvUJ5SVVVmO1vhNmQIkO6tA6OUpbF3q7qbB-VDfKHJRqc0rbDCJU32OAkadEKD2GuL61QyhBkONj4YSOzR0bKz8IXcemts08IZgjKdd3Wfh-1wxyaNfEb5s7ZcS8I-Vu0_nkH6G24OZQ69szzI2l0x6p26bfWelkWjoA9D2tY7rEKh1ah0t5fT0FSWQd314naRH51ptbtXopANMUe9ymgTZBwSMTbhhiwWAtMAGXzSNMuspmAQkpYyGYrlKEboA8N2S2q-4sRUsZia4Z2kY6dTTSO6h9hsWdhdg0ugctw_sS9eV1BFXbOBSujahNAmZTx7UFksLmKlQyDJkDPppSSLqhohLIHHzf0MccQ6EPHrCLPEHCDg6FO4o2jEYJ8xlGGnu10pIgwb87KlkXlRe7F2_AXooVDAtcpcmptjWt_Xiu768p8dnGIsJCeAESImRM-Tx0vUC5thfaKpTK8zihGAdeyIRNSOAFOAxtF2EnVJ5xpUQfiw0lctZGKW3s4N0gyC3HOi9v56gth8VnYkEFLUU63y2ZLfiqMsQ6ALXKm-n8ol6rq8xYcpgsNT2WAJBxni6gPAMosNy6VrNfX26daJshXxsOM3D0BXOt5hM8NHErAv_Ij5ICfku9o22dgL5fUtdkcB6AD_AJ7Ob_NGU_xB98gD9KaWL9mN5OfoFfAVhad1lDChJIfYWV7waeDpWFIbKRh12fB8AzjsKE-cAutu0g23E8hAQmDpa-IxlVhC4VMM_wD9Z-bWqa3aiFKb2wrs2xsoO5zgnRG4Ow9pNnzoIAmx3PqnBcCC0RYP869DG0cmLQJzxVxRRwcq4vEpks7ke7u3qMS1sKEVDl-B5yleeQMPA8xkPJsOOoICCByxCsLmEIfvm-p8DP8GywdkLCSVfFskl5P_xLy8jT7lrup7MZYF3pxyl0SlI-ja4tnSsCcodhYt3ILIvSLLfmiU5Y-kmms0nFuDmsDKBEgzYSbj9NYAKgdizv1sRBfpLZPJocg4xIs8j_pxpA9gAjHc7jKxN7mEaJVneN0m6vBReCCAEop8gOmeciMCul47hCEZ9SiTxq8yDgDAehcAMKNiiVLhIKGC5gju121wL3OQk9mP6f6W9LRfo_E-HOAMLPp6lKoq_WfqbD5yuUu-36oc-JUjqi6StQ4z5mOAT8ggsgXAcIXiLMQXppgYaVgwLhK7A1bc09oivBbNZS7gz9-yn3NhWNK1iQO_lMakv_WMms2IzW_qgvRbqGGQTWgcqLrLzX97_q27jgG3J_URD6AnPlMdvnwvaox3ngS5uDsxBi-NtWngp9h0rlUlswyZDrkdAj4HdJicO-Qui6v3yl-zvkai0RTf8aTi9GK9GfRTM93SWemg-mE9KbK2EIPlsAdpJnY99Gvq8kAqVAwJXwAmqHHveZvkNf0QBjJezAYX6Au2LI4d1oa40ny5w9tXyZgCNueQoQG0I1_aVaAcKaRTI5gctCFN_B7Xti8wnjQfvpoyz8abOF9uZWb6ssZNodFeJgj4cedYgjPSpJKDh40DYD28l3hUOZTVwMTOS5CoGHzRizgVs4J572GkG5dNdOTE6jWaSXpEg123QUORtCeJkaZ5W5ccNVdJrR8BezWhh1tkt0iqwJjZd3FCyMcHmld2eKRTheEwTYYrkCNaSHq9_0glUubXJp9lZKG21xomKCO7vCnWFbi2vwLZ34OwCkyaef6E0iS9MecNJZaxqWhKX6WmSyLu12cRdodTj830yJDBL68ObRezCfd4M0Ayk16UipjpvgIrCFFHepExBQES4iCtnC4wQT5YQuAQvJAZOIc3AgQEEQhF3F3BCq-4HP72iFJfSNMe6QiFkrvc5ty6dvovgpmKRRdXPdQtKZA_Zls_vmDgzij94vp4RTXzk4wDZmRMfZAh4wV7AQeWA6-gEKJA6kAAnBBMWcgBuGoQkCoxNwTH17PKdErJ9TQjryWb_kpVFZbXv-GdNKMF-2g-fS_6aVPDKtBLNBMj9JgugKjO3juQdq0CpvtismHfGqSZNO2mnae7AO1XM2IHiBvXdsMBrbXOIS7WfpZ5QC6Upwct0AlKPvKqX3VAQXgcJIeYhRh4fMYcwLXIdyCqxlSwYqrTz2ZOnxT97EweJnT6GOJmRddNmnF3_CthjVVxNnVJlWmmyhobTqBdWjjPWkE1IsWWork2oxeZsmz7VWBQerLK4VYZils7JSrv0f44hVH6FL0HVGqfMO0Rte0wZmSethmllK-lNLs1Sh3_wq-wAOhe4AGPxffNvLoOjsi45B0HMxDN4FtW3Nr7XdB7XRD38qjhrkBj7IDWAO5oscnCNAWnGTxuGk_mW9B2oDWzHpeFahTUAhOmAHCjdEvmQ0EA6hgmAPozAIMJde4ISYOlL5WCmpBJWYcYUQB3egb9-39hDIWJghjlMfjCnjj8-0NVd6HZT2Ug0XQVLzHJHSxpzFGtLV_FUK_bofC5AZhRF8ABpJomIOnrmJTwTpXJ_j-1dxzMRAcOgAVmlqvZVRPDlWsVEfFzps3M6FsxUjmLqeZJjrUDa4QS6R3EHItgV2qRJg3zD9WIkAM1_ngHLwjBUPqArB9r9j7zQSpuOYsclzYpXpWjVGNZ_H6Y1Wktgqs7MabC8-3VkHEMmlS7dXGutXUIl3kx16oJqF62x_8nL7xrxU0953eeZ2AujwN77jqD-zyZ2905oYnxHSbOhoZ7DfRT65hEkV_3FaeHVmxSBpD8U9DzRuS908eX-TPjebm3eo2_NCl3uuHYaSeipwXEZZKGww7rErzMaN77vYcRzue7YKXduXSNkM-27gssBxluyr8675-QyL3pqnxtFrlnw0lmAkT-uVW220Alygs9tGvU2eCTFAhaxNhUZHPrPxwK7-qp3Khnap6BD_f0OOME131J88-ZKC9fJen6yZXIKtXeewHsu5p7JPk_IonE4TZtXvu5Yi5YT07UuXrzAvKUjrTmzf90NGXKYCP1TgtzLqMxwoGQD9Y9_G4GtxH-xNasuQeHZIghBIH4UhAmEODm9_T2XQjjx6fgSW5FHflrR7OzCC3iVVitpk2Q1WXBsLElVMULEEyF1-h5JbapsYtX0XUKW_S5m7yHjLrUIf7CjtTGPUeQrMPVX29Gci8sFUpMGo7IH0wEBp3ZNwLf_oJlW_y8CL6FgDkgvmKeRwhwsfE-QGnlQekZQqkImOqzDltudIQSnnREjJkYd4iGH1dGi2b7kNZ0-zsexpGz0iX5ridfOla4gtaqJDRuAaydP3WOInllIED-3HpvPYeufP4znYHObwkpnPIiets_bY5SAImId8WG3wOTkSGPxTIrhrB0Lq4K7nK4eKANvMpwJcJaKFh04QxMpboit7CdGd_bmViu838L6aIP7-PNOtTJTmEkShkS0D2YPPKB9Qg7i0-LRq1jq6CZu0deHGDPqn5no6tB-sEw9maRLBpI_TNJzU96KYCytrfYQJHdVHGP5zJ96xQh-1KSlktpTEp44rQuLbDOwvJihn1AUao0BMkit95MaV2LM9QlwUcPAlKA0DoRyP9n2K1vmaVtL9nrVn7bapjHWpzO3GCsoQXjf0PjnI0mu9M5sCY_Sj-rP0C1DGtnmLTpMIrYRCz4nAHXPp7qcOVbY_D3gQxzK3sLXr3ea5XASAeyKse7pjYtK8mxR1cxAW2MSEh2EW2j-tpd2SuodZ-odK_psLviGPhfQii2_fXZztnk6OQJkYaXOoiiifdkLojuMr4fkITDUundCVAaJcEIKJjSViWMfTlWsjYnsi5Jwj5CMmiKc88NW9ruKlfJxl2ifUei4M7oT5Gj-hCik_dDeBDAeWzgCmp3doJ_rJnVvw5uPOJjEOiC2p64QCDItAEg-BKRsGIWMCSxtQ4wWEhNwV0vaZjaTePg5somhge5j3su2cjvnR26uvVEXj73Njqh6or4UyO2bP9FGsgQTNZxpjC33SKi-3Y6orExZRzUaFfch1po4Ki5kOYDJruHpaxknjW_00ZWySv6LFmbjqihTdt96lHgBQV2m6zcyJwT_JgY8lgUoiBpkLlhRWS4vNT6objuS-VGClEheohQSO53rUUTSkoXQRlsyRNkUe5Z4PbIVthQJwkgLfwZ6wsY1JNwhg40He0k7R3mjaPZ7QjmVb3gg2svX8cEYbPgoO5sw898usozdx9DXKJhdzr6KTztYnwo4L1jwnnEhOPZ0l6SMqXNfjQYAdqVPCQtch3HaQcnwOZqKSoMSpkp6r2Og2HlgTDWaw08VMrZ4-GLtusXfwwewdHM1lFpSGYE8P1hZ-x88rnbwgVdpPAKq_1U8-9nzEyqZX3XO0OjTizQtLR5GMF1AdqC2_B1asyyIdxFu4DGZbogvabIWQ2k54QtPPGTxI6ylwfsw5sYXT9yOYfo3P914l0o_Sea4vlOukNnvEo4wLlypEiG2DPxcQR2L9JGqgj5kopDMHHewwqSjWBptwJGUEKIIovqbf17fQNuP3tWwqKgZcwNb3QW9QbyrtD1pF-oxlou7lBf7Tt3HIcKxq4MrESe_GxNo1IGzcNWDcEX3XgLkrQ1WssxOKEJGeA-aL4mDw-6GSYPEAJXFKHG3qgEYHMewzTWiOTZCDbBeMIuGGDAtEUWcndGz7k7ejVneO6PbEs_Eq61wf41zevYFxccDT6e4pGUwZd7K7afqMWYub7LSzoSno2oiHrp-Km6ADa9scuAxTleVNGup4UlEZRGuqdrzs6kat9kBwNy2hW_9MxUr1kw4WzZpRfv-9rz93ygFdJ1BXnyHsxOrMbYyyY9dIoh0EFDDhEPgteQhMwjwFnEKpkJKhwAkCNwy0_rbhpy1920MKjGrwtPmo0Eajsd0nkNnEfYDMFmgJcf-ZgnaUrnUjxs_zyP-UD1yL8UZexZ2sZsltwsE3dGmIZIiZ6wQ08MDq5YqDAg9sESIfCCUkIcaCIsYZWMGecGxKucN5P4o3TBRPosjb6bgrLrygYxcI_al0NR1OQDLZt6XpC5bcRL_PCsT9k74DplItdRDPJeNBPJu6dzQ1WqGpiQB-74gYJH3qUKr3ScGNchXGPihnEoADEFDPDTDivnL9gAWBwD5HoQceOjgFPgrAx1pmF7aTkzpxif7lKuNJvsIE7ftayuzx6IvSKnfAZA_duVxCG3umTxNqXiT26QvXLX3bdpPrp_8qc420jq-2qEqlrXOZ9BkNk5ubyF4bkzLUdfyGg92LzbF2xIGv2LD9z9W8PU7S72iPX5bWfWLbLM9J8OHitP_gennfdZl423qD_dyDuq-3FkO7jMGpbJdX7ZpntusH1OmqNjre1Qv1Wq0X09G6ffZh8PvDIGvBGEBe-Rh8G3fN8_D3RJ15Yqc_DrEBfIgN4OM-MMw2Sx8AvQeA8pHnPgR8nyEMkNaD0DC0JnwDa4I3AIPdEyFNdcIfRZSbYNL1YJS3Qi5ujiwhltZfaxEofRBFkkfSE33kDFrLR5xVAygDFI-homGmEmvKOfCmKO4KuqroHpJu8Xh7rSLYQ_DWmjSxH4a3-3Dw7mwedwbNH0rydANymK65XgS7DuuuV1U0tF5DCy4E7QGoitYEQFwbk94IyqJ7UMzw8rFHcl6L88nKldAJle0W2LmX8mh1JdabbauF-yDBxh-JnhaN0QeKFfJowfR4e4E91mRhD1FzQ_qS3msk9cvym5f2fFMmEF5TDLkMOV0ZUJbcQwQsQQhZ10hn3KE9K70sutc4Bkjk3mZxi7npw4Sf2Awx3YcUhgdyH6o2QrTtHIoHobut-vFDVejG2JpvZiXEI-1Q9uiVfKwljDeDCLop6bQxQA9xGVvKk90DGS3ecB9FlRtajAdzSYtF2VL7UGBX9OzDsmhNA9N2RS_0Upas2VxgBpjuKoWy6D5KYYAbNqZk11VvDBPRU29V0f2jUGQDvj7ZQLBjGCMbE913AZUx1fIlQf3chQxUmS3-UenIeV6-h2HeqjAPDKZJGF3deS5QR2DNhfzvktZdGWVS7ZblpWmsZFLZ3HcfG9TJM_oON_NMTf3YzuLRQTbQAnzTQl-ToTLzbN9t9cBN_U4hQmjoUcPoKorfhabyZSFX9lLPak8p_aja4vjv-jOqX_5Za1ZK3-99ku-l-jXJxSuN51GSSD8e6kU_6bib-TJRl_NMP260aljXMimmKk3O5O9pdpQGTS-n80RmUb4MCZdToKFyd601jSE8-_O8SGdnadCaBv_BOhnctxtoP88HELdqbtCoSz-rGtQT-6DfnTYu4D0ambvY3yWLBPJ1Sb1uf5relK_Q3bfHB_BXvexR0lv2I5nNP01lsrJfs_hqjb4aztT7YubYfMM2NX-ShkHPY-krfZO9ytpsUNJG3cJeVd8kA1aP19Wt6IpG9ebg4pnRRUNCGF7RX1a_RVm34sub6CTi_gj5qmlByf0Hp3u6fyuzJ9o0cNho9Zb4jfTLWSq4NA-a6gdtL1Uctlaar57ffXFSr9q5fnOqXus1G917BTRh9ZHpuGL1vO4zPL1i96lfDugijWOzY9pmlOUNF1b3or4gjrsGqfcRsIqzFjvz96fC-oHIRpljNo6NZTodL17zqoyWVy_OZTF9F-7NI3OJ7l8n_x-Bw7TM
```

### Suggestion

This issue seems to have happened already (https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues/6519). I suggest to simply ignore the hit chance when calculating the trigger rate and instead multiply the final effective dps of the triggered skill by the hit chance.

**Comment by Paliak** [comment]
This is an issues with basically all "chance" like modifiers (trigger chance, crit chance, accuracy). I've discussed this with other devs after seeing this issue: https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues/7232

I left the patch (https://github.com/PathOfBuildingCommunity/PathOfBuilding/pull/7244) as a draft for multiple reasons:

The suggestion was to apply chance modifiers after the calculation of breakpoints (just like you've suggested basically but as a probabilistic scalar instead) to the calculated trigger rate. Possibly due to my lack of understanding of the problem that didn't sit quite right with me as in my mind the trigger sequence goes:
attack -> hit(%) -> (crit)(%) -> trigger(%) -> triggered(%)

Why would hit and crit probabilities be applied after triggering?

Additionally i wasn't quite sure how to apply the  suggested approach to setups with multiple skills.

Before my "rework" the chances were applied to the final rate of the triggered skill as a linear scalar which i then moved and applied to the attack rate instead.

The issue you mention was additionally affected by changes to the function calculating breakpoints in a failed attempt to make it faster and better handle specific edge cases. The change has since been reverted. I'm still not sure whether it is correct or not and honestly i've heard some many theories of how this should all work at this point that it will be hard for me not to doubt any implementation.
